sources/web/datalab/polymer/components/datalab-app/datalab-app.ts

```diff
@@ -117,15 +125,40 @@ class DatalabAppElement extends Polymer.Element {
    * We do this to lazy load pages as the user clicks them instead of letting
    * the browser pre-load all the pages on the first request.
    */
-  _pageChanged(page: string) {
+  _pageChanged(newPage: string, oldPage: string) {
     // Build the path using the page name as suffix for directory
     // and file names.
-    const subpath = 'datalab-' + page;
+    const subpath = 'datalab-' + newPage;
     const resolvedPageUrl = this.resolveUrl('../' + subpath + '/' + subpath + '.html');
     Polymer.importHref(resolvedPageUrl, undefined, undefined, true);
 
-    // If the new page has a resize handler, call it.
-    this._resizeHandler();
+    const newElement = this._getPageElement(newPage);
+    const oldElement = this._getPageElement(oldPage);
+
+    // Call proper event handlers on changed pages.
+    // TODO: Explore making all datalab pages extend a custom element that has
+    // these event handlers defined to keep things consistent and get rid of the checks.
+    if (newElement) {
+      if (newElement._focusHandler) {
```

Comment:
I think code like this inner if-statement is often written as a one-liner:
  newElement._focusHandler && newElement._focusHandler();